wgsl/index.bs

```diff
@@ -9133,6 +9133,138 @@ Cube faces are square, so the x and y components of the result are equal.
 If `level` is outside the range `[0, textureNumLevels(t))` then any valid value
 for the return type may be returned.
 
+### `textureGather` and `textureGatherComponent` ### {#texturegather}
+
+A <dfn noexport>texture gather</dfn> operation reads from a 2D, 2D array, cube, or cube array texture,
+computing a four-element vector as follows:
+* Find the four texels that would be used in a sampling operation with linear filtering,
+    from [=mip level=] 0:
+    * Use the specified coordinate, array index (when present), and offset (when present).
+    * The texels are adjacent, forming a square, when considering their texture space coordinates (*u*,*v*).
+    * Selected texels at the texture edge, cube face edge, or cube corners are handled
+        as in ordinary texture sampling.
+* For each texel, read one channel and convert it into a scalar value.
+    * For non-depth textures, a parameter specifies the channel to use.
+    * Depth textures only have one channel.
+* Yield the four-element vector where the components are taken from the selected texels with
+       relative texel coordinates as follows:
+    * <table>
+        <thead class='data'><td>Result component<td>Relative texel coordinate</thead>
+        <tr><td>x<td>(*u*<sub>min</sub>,*v*<sub>max</sub>)
+        <tr><td>y<td>(*u*<sub>max</sub>,*v*<sub>max</sub>)
+        <tr><td>z<td>(*u*<sub>max</sub>,*v*<sub>min</sub>)
+        <tr><td>w<td>(*u*<sub>min</sub>,*v*<sub>min</sub>)
+        </table>
+
+```rust
+textureGather(t: texture_2d<f32>, s: sampler, coords: vec2<f32>) -> vec4<f32>
+textureGather(t: texture_2d<f32>, s: sampler, coords: vec2<f32>, offset: vec2<i32>) -> vec4<f32>
+textureGather(t: texture_2d_array<f32>, s: sampler, coords: vec2<f32>, array_index: i32) -> vec4<f32>
```

Comment:
so these are the same as `textureGather0`, right? Do we really need these extra overloads then?

Reply:
Yes, it's the same.   It's a design choice.

I went with the green light from early review: https://github.com/gpuweb/gpuweb/issues/2327#issuecomment-976560843

Reply:
Reworked according to feedback from this week's meeting.